### PR TITLE
[HOPSWORKS-1372] Upgrade the ELK stack to 7.2 OSS and use the OpenDis…

### PR DESCRIPTION
**Pull request by maismail**
…tro Security Plugin

## Make sure there is no duplicate PR for this issue

* **Please check if the PR meets the following requirements**
- [ ] Adds tests for the submitted changes (for bug fixes & features)
- [ ] Passes the tests
- [x] HOPSWORKS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit

* **Post a link to the associated JIRA issue**
https://logicalclocks.atlassian.net/browse/HOPSWORKS-1372

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:
tests: https://github.com/logicalclocks/hops-testing/pull/565

### CODE REVIEW

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticClient.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+import java.nio.file.Paths;
+import java.security.GeneralSecurityException;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Singleton
+@TransactionAttribute(TransactionAttributeType.NEVER)
+@ConcurrencyManagement(ConcurrencyManagementType.BEAN)
+public class ElasticClient {
```
Comment (tkakantousis) [inline hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticClient.java]: @maismail we can add a dependency on the other singleton, Settings, like here https://github.com/logicalclocks/hopsworks/blob/master/hopsworks-common/src/main/java/io/hops/hopsworks/common/airflow/AirflowManager.java#L91

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticController.java [hunk 2]
```diff
@@ -277,24 +240,22 @@ public void updateExperiment(String index, String id, JSONObject source) throws
       projectSearchInSharedDatasets(client, projectId, searchTerm, elasticHits);
       return elasticHits;
     }
-
-    shutdownClient();
-    throw new ServiceException(RESTCodes.ServiceErrorCode.ELASTIC_SERVER_NOT_FOUND, Level.SEVERE);
+  
+    //we need to further check the status if it is a probelm with
```
Comment (tkakantousis) [inline hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticController.java]: @maismail probelm typo

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticController.java [hunk 3]
```diff
@@ -396,92 +380,61 @@ public void deleteProjectIndices(Project project) throws ServiceException {
...
+        Long creationDate = index.getLong("creation.date");
+    
+        if (pattern == null || pattern.matcher(indexName).matches()) {
+          indicesMap.put(indexName, creationDate);
         }
       }
+      return indicesMap;
+    }catch (IOException e){
+      throw new ElasticException(RESTCodes.ElasticErrorCode.ELASTIC_INTERNAL_REQ_ERROR,
+          Level.INFO,"Error while getting all indeces", e.getMessage(), e);
```
Comment (tkakantousis) [inline hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticController.java]: @maismail indeces typo

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticUtils.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    // /security/configuration/PrivilegesInterceptorImpl.java#L276
+    return Settings.KIBANA_INDEX_PREFIX + "_" + project.hashCode() + "_"
+        + getProjectNameWithNoSpecialCharacters(project);
+  }
+  
+  public static String getAllKibanaTenantIndex(String project){
+    return getKibanaTenantIndex(project) + "*";
+  }
+  
+  public static String getProjectNameWithNoSpecialCharacters(String project){
```
Comment (tkakantousis) [inline hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticUtils.java]: @maismail where is this used?

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/KibanaClient.java [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+import java.security.GeneralSecurityException;
+import java.util.Base64;
+import java.util.logging.Level;
+
+@Singleton
+@TransactionAttribute(TransactionAttributeType.NEVER)
+@ConcurrencyManagement(ConcurrencyManagementType.BEAN)
+public class KibanaClient {
+  @EJB
+  private Settings settings;
```
Comment (tkakantousis) [inline hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/KibanaClient.java]: @maismail add dependency as in the elasticclient

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/KibanaClient.java [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.GeneralSecurityException;
+import java.util.Base64;
+import java.util.logging.Level;
+
+@Singleton
+@TransactionAttribute(TransactionAttributeType.NEVER)
+@ConcurrencyManagement(ConcurrencyManagementType.BEAN)
+public class KibanaClient {
```
Comment (tkakantousis) [inline hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/KibanaClient.java]: @maismail where is this client used? Can we add a comment in the class description?

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticClient.java (hunk 1 above)
Comment (maismail) [reply hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticClient.java]: Fixed

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticController.java (hunk 3 above)
Comment (maismail) [reply hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticController.java]: Fixed

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticController.java (hunk 2 above)
Comment (maismail) [reply hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticController.java]: Fixed

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/KibanaClient.java (hunk 5 above)
Comment (maismail) [reply hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/KibanaClient.java]: Fixed

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticUtils.java (hunk 4 above)
Comment (maismail) [reply hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticUtils.java]: It is used in ElasticJWTController 
https://github.com/logicalclocks/hopsworks/pull/405/files#diff-f04918739b3b486d334518ca1e426a3eR95

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/KibanaClient.java (hunk 6 above)
Comment (maismail) [reply hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/KibanaClient.java]: It is used in ElasticController to make calls to Kibana 
https://github.com/logicalclocks/hopsworks/pull/405/files#diff-3bd4736a4063c172cef761d092400ea5R122

**Pull request merged**